docs/examples/tutorial/cython_tutorial/primes_cpp.py

```diff
@@ -0,0 +1,22 @@
+# distutils: language=c++
+
+import cython
+from cython.cimports.libcpp.vector import vector
+
+def primes(nb_primes: cython.uint):
+    i: cython.int
+    p: vector[cython.int]
+    p.reserve(nb_primes)  # allocate memory for 'nb_primes' elements.
+
+    n: cint = 2
+    while p.size() < nb_primes:  # size() for vectors is similar to len()
+        for i in p:
+            if n % i == 0:
+                break
+        else:
+            p.push_back(n)  # push_back is similar to append()
+        n += 1
+
+    # Vectors are automatically converted to Python
+    # lists when converted to Python objects.
```

Comment:
This is worded quite weirdly; perhaps the following or something similar?
```suggestion
    # Return values are automatically converted to Python objects,
    # in our case the vector will be converted to a Python list
```

Reply:
More words, harder to understand. Do not forget, this is, in theory, the first thing that those who want to start using cython see.
But I agree, two "converted" in one sentence doesn't sound very nice.

Reply:
This isn't just about "return values", so the change might give beginners the wrong idea that returning something from a function leads to special behaviour.

Reply:
Perhaps the following is better then?
```suggestion
    # C++ objects get converted to Python ones, this means
    # the Vector will get converted to a Python list
```

Reply:
> Perhaps the following is better then?

This improves your suggestion a little more: "C++ objects are automatically converted to Python objects, so the vector will be converted to a Python list."

But then again, more words explain the same thing.

Also, I don't really like the fact that we are only talking about C++ objects. It seems to me that this introduces a second ambiguity for understanding. Some might understand it like this: Only C++ objects are converted, and the rest, like C objects, do not.

What do you think about it @scoder?

Reply:
I suggest even more words:
"If possible, C values and and C++ objects are automatically converted to Python objects at need, so here, the vector will be copied into a Python list when it is returned.

Reply:
For context, this is a comment right before returning, so I think this would be sufficient:
```python
# If possible, C values and C++ objects are automatically
# converted to Python objects at need,
# so here, the vector will be copied into a Python list.
return p
```

Reply:
@scoder resolved [here](https://github.com/cython/cython/pull/4226/commits/aff7601737563d476e43e01e20e86071692bec7d)